pkg/controller/kubedirectorcluster/cluster.go

```diff
@@ -306,6 +333,33 @@ func (r *ReconcileKubeDirectorCluster) syncCluster(
 	return nil
 }
 
+// Calculates md5sum of resource-versions of all resources
+// connected to this cluster
+func calcConnectionsHash(con *kdv1.Connections, ns string) string {
+	clusterNames := con.Clusters
+	var buffer bytes.Buffer
+	for _, c := range clusterNames {
+		clusterObj, _ := observer.GetCluster(ns, c)
+		rv := clusterObj.ResourceVersion
+		buffer.WriteString(rv)
+	}
+	cmNames := con.ConfigMaps
+	for _, c := range cmNames {
+		cmObj, _ := observer.GetConfigMap(ns, c)
+		rv := cmObj.ResourceVersion
+		buffer.WriteString(rv)
+	}
+	secretNames := con.Secrets
+	for _, c := range secretNames {
+		secretObj, _ := observer.GetSecret(ns, c)
+		rv := secretObj.ResourceVersion
+		buffer.WriteString(rv)
+	}
+	// md5 is very cheap for small strings
+	md5Sum := md5.Sum([]byte(buffer.String()))
```

Comment:
It seems like we do want to include the resource names in this hash too. We don't want to miss an update if configmap foo is switched out for configmap bar and they both happen to have the same resourceVersion.

Reply:
yes good point. Done